core/src/test/java/org/apache/iceberg/TableTestBase.java

```diff
@@ -236,12 +254,22 @@ ManifestFile writeManifest(DataFile... files) throws IOException {
   }
 
   ManifestFile writeManifest(Long snapshotId, DataFile... files) throws IOException {
-    File manifestFile = temp.newFile("input.m0.avro");
+    return writeManifest(snapshotId, null, files);
+  }
+
+  ManifestFile writeManifest(Long snapshotId, String compressionCodec, DataFile... files)
+      throws IOException {
+    File manifestFile = temp.newFile();
```

Comment:
It uses `InMemoryOutputFile` in `TestManifestListVersions`. Or maybe we could put those UTs to `TestManifestListVersions` to reuse those test util methods.

Reply:
Firstly, `TestManifestListVersions` is about testing versions (v1 versus v2). Secondly, `TestManifestListVersions` has a `writeManifestList` method (which uses an `InMemoryOutputFile` as you observed), but it doesn't have any `writeManifest` method. So there is nothing in `TestManifestListVersions` to reuse for this, and the tests I'm adding around compression in manifests and manifest lists do not fit the purpose of `TestManifestListVersions`.

I don't think it's an issue that we're actually writing to disk, is it? There are other test classes calling `writeManifest` (not just `TestManifestWriter` and `TestManifestListWriter`) and I didn't check if and how they might be affected if we change this to using `InMemoryOutputFile`.

Reply:
Both the `InMemoryOutputFile` and `FileBasedOutputFile` should work as expected. I think the `InMemoryOutputFile` usage is aimed at accelerating the testing. Hold on to your thoughts. It is not a great deal.